updates/P8.txt

```diff
@@ -0,0 +1,136 @@
+finalized on the chain) and decrease the throughput (that is, the maximum number
+of transactions the chain can incorporate in a given period of time).
+
+The second role of validators is to sign off on rounds by either attesting to a
+valid block in the round, or to the fact that the round timed-out. To progress to
+the next round, two thirds of the finalization committee (weighted by stake) must
+agree. If more than one third of the committee does not sign off on a round, then
+the blockchain halts until two thirds of the committee eventually agrees.
+
+Inactive validators are therefore detrimental to the operation of the consensus
+protocol. Protocol version 8 introduces a mechanism to automatically suspend
+validators that consistently fail to participate in the protocol. Suspended
+validators are excluded from the committee of validators, and will therefore
+not be selected to propose blocks nor be required to sign off on rounds.
+Consequently, suspended validators do not earn rewards for themselves or their
+pool.
+
+The account associated with a suspended validator can lift the suspension by
+sending a transaction to do so. The validator will then be included in the
+committee the next time it is determined. (The committee is determined one
+epoch before each pay day.)
+
+The procedure for automatically suspending a validator is as follows:
+
+- For each current validator, continuously track the number of rounds in
+  which the validator was the round leader, but no block was produced.
+
+- At each payday, if the number of missed rounds for a validator exceeds a
+  threshold (determined by the "maximum missed rounds" chain parameter)
+  then mark the validator as "primed for suspension".
+
+- At each snapshot (that is, the start of an epoch prior to a payday), if
+  a validator is primed for suspension, then it is marked as suspended.
+  Suspended validators are not included in the calculation of the validator
+  committee.
+
+- Whenever a validator produces a block or signs a quorum or time-out
+  message, its missed rounds counter is reset and it is no longer marked as
+  primed for suspension.
+
+                               Changes
+
+The following behaviours are changed in protocol version 8.
+
+1. A new chain parameter, `maximum_missed_rounds`, is added. This is updated
+   by a level 2 update instruction (payload `ValidatorScoreParametersCPV3`)
+   that is authorized by the pool parameters keys.
+
+2. The validator record on accounts includes a new boolean flag indicating if
+   the validator is currently suspended.
+
+3. The validator pool reward details add a counter of the number of missed rounds
+   and whether the validator is primed for suspension.
+
+4. On executing each block:
+
+    (a) Validators that signed the quorum certificate for the parent block
+        have their missed-round counter reset to zero and have their primed-
+        for-suspension flag cleared.

```

Comment:
This description doesn't correspond to what I had understood from our kick-off discussions. What I had understood is that at the snapshot block, a validator would be primed-for-suspension if their missed block counter was > threshold AND if they hadn't gotten any signatures in QCs/TCs **that day**. The result is that a validator would have to be offline for 47 hours before getting suspended: 24 to not get any signatures in QCs/TCs and be primed-for suspension, and another 23 of the same until they are suspended. But this description implies that if a validator goes down 5 minutes before payday and misses enough blocks in those 5 minutes, then they will already be primed-for suspension. 23h later they will be suspended. So we are really looking at a scenario where 23h of inactivity might be enough to be suspended. Correct?

I actually like this version more than what was initially discussed. My comment here is to make sure that I have understood this right, not for anything to be changed.